Add Rails 7.1 support

camertron opened the pull request:
### What are you trying to accomplish?
<!-- Provide a description of the changes. -->

This PR makes the changes necessary to support Rails 7.1.

### Integration
<!-- Does this change require any updates to code in production? -->

No changes necessary in production.

#### List the issues that this change affects.
<!--Every code change must address _at least 1_ issue. Fixes a bug, completes a task, every change
      should have a corresponding issue listed here. If one does not already exist, create one. -->

Fixes https://github.com/primer/view_components/issues/2318

#### Risk Assessment
  <!-- Please select from one of the following and detail why this level was chosen -->

- [x] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### What approach did you choose and why?
<!-- This section is a place for you to describe your thought process in making these changes.
     List any tradeoffs you made to take on or pay down tech debt.
     Identify any work you did to mitigate risk.
     Describe any alternative approaches you considered and why you discarded them. -->

The only thing necessary to support Rails 7.1 is to remove an unnecessary `require` statement in the Rails engine. Rails 7.1 no longer adds eager load paths to Ruby's load path, preferring instead to let Zeitwerk autoload missing constants.

### Accessibility
<!--
  You may remove this section and the "Accessibility" heading above _only_ if the changes in this pull request do not impact UI. Delete all those that don't apply.
  If there are any accessibility-related updates, please describe them here.
-->
- **No new axe scan violation** - This change does not introduce any new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violations.

safeforge commented:
@camertron Removing the `require "primer/form_helper"` does not appear to be sufficient, but you have correctly identified the issue.

I'm experiencing issues with `Form` usage, and the error appears to be similar to the previous one: "cannot load such file -- primer/class_name_helper."  

Indeed, by removing `require "primer/class_name_helper"` from the file `lib/primer/forms/base_component.rb`, the issue with my application appears to be resolved. I wonder if there are any other `require` causing errors.

camertron commented:
@safeforge interesting, thanks for the heads-up. Let me see if I can get our test suite to fail, that would really help catch these issues.

camertron commented:
Ok, I totally forgot to bump the framework defaults in the demo app's application.rb. Doing so uncovered several additional bad `require` statements that I have since removed.

@safeforge can you verify this addresses the issues you're seeing?

jonrohan commented on .github/workflows/test.yml:
```diff
@@ -51,6 +51,8 @@ jobs:
             ruby_version: "2.7"
```
How do you feel about also dropping this matrix?

camertron replied:
I'm ok with that 👍

jonrohan replied:
I like the simplicity of having 3 groups, and adding a new one makes me want to drop the old one. I don’t know what sorta social contract we have to the community to always test against 6.1. I say let's go for it cause it lets us move our dependencies forward

camertron replied:
Decision: leave 6.1 support in as long as it's easy. Remove it as soon as it's inconvenient.

jonrohan reviewed (approved):
👍🏻

safeforge commented:
> @safeforge can you verify this addresses the issues you're seeing?

@camertron, I can confirm that your fix has resolved the issue in my use case. Thank you!
